packages/plugin-ext/src/hosted/browser/hosted-plugin-controller.ts

```diff
@@ -55,34 +57,51 @@ export class HostedPluginController implements FrontendApplicationContribution {
     @inject(HostedPluginLogViewer)
     protected readonly hostedPluginLogViewer: HostedPluginLogViewer;
 
-    private pluginState: HostedPluginState = HostedPluginState.Stopped;
+    @inject(HostedPluginPreferences)
+    protected readonly hostedPluginPreferences: HostedPluginPreferences;
 
+    @inject(PreferenceServiceImpl)
+    protected readonly preferenceService: PreferenceServiceImpl;
+
+    @inject(MessageService)
+    protected readonly messageService: MessageService;
+
+    private pluginState: HostedInstanceState = HostedInstanceState.Stopped;
+    // used only for displaying Running insted of Watching in status bar if run of watcher fails
+    private watcherSuccess: boolean;
     private entry: StatusBarEntry | undefined;
 
     public initialize(): void {
         this.hostedPluginServer.getHostedPlugin().then(pluginMetadata => {
             if (!pluginMetadata) {
                 this.frontendApplicationStateService.reachedState('ready').then(() => {
+                    // handles status bar item
                     this.hostedPluginManagerClient.onStateChanged(e => {
-                        if (e === 'starting') {
+                        if (e.state === 'starting') {
```

Comment:
Should we have State defined through constants ?

Reply:
Sure, will fix it